Cargo.toml

```diff
@@ -106,7 +106,7 @@ libp2p-plaintext = { version = "0.38.0", path = "transports/plaintext", optional
 libp2p-pnet = { version = "0.22.2", path = "transports/pnet", optional = true }
 libp2p-relay = { version = "0.14.0", path = "protocols/relay", optional = true }
 libp2p-rendezvous = { version = "0.11.0", path = "protocols/rendezvous", optional = true }
-libp2p-request-response = { version = "0.23.0", path = "protocols/request-response", optional = true }
+libp2p-request-response = { version = "0.24.0", path = "protocols/request-response", optional = true }
```

Comment:
Given that this is a breaking change for `libp2p`, this pull request needs to bump the `libp2p` version to `v0.51.0`. Can you please also add a changelog entry to the root `CHANGELOG.md` file. See previous entries for reference.